genienlp/train.py

```diff
@@ -481,6 +481,14 @@ def init_model(args, numericalizer, context_embeddings, question_embeddings, dec
     return model
 
 
+def get_linear_schedule_with_warmup(i, dimension, warmup, num_training_steps):
+    i += 1
+    warmup = max(1, warmup)
+    if i < warmup:
+        return float(i) / float(max(1, warmup))
+    return 1. / math.sqrt(dimension) * max(0.0, float(num_training_steps - i) / float(max(1, num_training_steps - warmup)))
```

Comment:
Do we want an option for this? Or is this debugging code / experimental code that is unused?
In the latter case, kill it.